File: packages/transform/src/features/json/JsonCreateValidateStringifyProgrammer.ts
import { JsonValidateStringifyProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonCreateValidateStringifyTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "json.createValidateStringify",
      write: JsonValidateStringifyProgrammer.write,
    });
}

File: packages/transform/src/features/json/JsonCreateValidateStringifyProgrammer.go
package json

import (
	corejson "github.com/samchon/typia/packages/core/src/programmers/json"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var JsonCreateValidateStringifyTransformer = jsonCreateValidateStringifyTransformerNamespace{}

type jsonCreateValidateStringifyTransformerNamespace struct{}

func (jsonCreateValidateStringifyTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	stringify := corejson.JsonStringifyProgrammer.Write(call.TypeText)
	return "((input) => ({ success: true, data: " + stringify + "(input) }))"
}
